native/src/boot/bootimg.cpp

```diff
@@ -653,12 +669,9 @@ void repack(const char *src_img, const char *out_img, bool skip_comp) {
     hdr->vendor_ramdisk_table_size() = 0;
     hdr->vendor_ramdisk_table_entry_num() = 0;
 
-    std::vector<std::tuple<std::unique_ptr<vendor_ramdisk_table_entry_v4>, format_t>> new_ramdisk_table_entries;
+    std::vector<std::tuple<vendor_ramdisk_table_entry_v4, format_t>> new_ramdisk_table_entries;
     for (const auto &[entry, fmt] : boot.ramdisk_table_entries) {
-        auto &[new_entry, new_fmt] = new_ramdisk_table_entries.emplace_back(make_unique<vendor_ramdisk_table_entry_v4>(), fmt);
-        memcpy(new_entry.get(),
-               entry.get(),
-               hdr->vendor_ramdisk_table_entry_size());
+        new_ramdisk_table_entries.emplace_back(*entry, fmt);
```

Comment:
here you always keep the ramdisk entry from the source boot image. how can the user remove or add entries?

Reply:
This PR is just adds un- and repacking support, which is clearly stated in the PR description.

Reply:
If you are only un- and re-packing, why copy all entries? Most of these entries (except those we cannot unpack) will be copied from files, right?

Reply:
Its a leftover from the previous revision. You are right, coping the entire is unnecessary. Storing a list of fragment sizes should be enough.

Reply:
To get rid of any copy operation, [writing the table entries](https://github.com/nilz3000/Magisk/blob/vendor_boot_v4/native/src/boot/bootimg.cpp#L819) would either require two `xwrite()` or to temporarily set size and offset in the mmaped entry of the source image.